src/App.tsx

```diff
@@ -14,11 +16,70 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const goods: string[] = [...goodsFromServer];
+
+  const [isVisible, setVisible] = useState(false);
+  const [isReversed, setReversed] = useState(false);
+  const [sortBy, setSortBy] = useState('');
+
+  const reset = () => {
+    setReversed(false);
+    setSortBy('');
+  };
+
+  if (sortBy === 'length') {
+    goods.sort((a, b) => a.length - b.length);
+  }
+
+  if (sortBy === 'alphabet') {
+    goods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (isReversed) {
+    goods.reverse();
+  }
+
+  return (
+    <div className="App position-absolute top-50 start-50 translate-middle">
+      {!isVisible
+        ? <button type="button" onClick={() => setVisible(true)}>Start</button>
+        : (
+          <>
+            <h1 className="text-center">Goods</h1>
+            <GoodsList goods={goods} />
+            <button
+              type="button"
+              onClick={() => setReversed(!isReversed)}
+              className="me-2 rounded"
+            >
+              Reverse
+            </button>
+            <button
```

Comment:
add an empty line between same level elements
```suggestion
            </button>
            
            <button
```